sdk/python/pyproject.toml

```diff
@@ -1,7 +1,7 @@
 [project]
   name = "pulumi_aws"
   description = "A Pulumi package for creating and managing Amazon Web Services (AWS) cloud resources."
-  dependencies = ["parver>=0.2.1", "pulumi>=3.0.0,<4.0.0", "semver>=2.8.1"]
+  dependencies = ["parver>=0.2.1", "pulumi>=3.0.0,<4.0.0", "semver>=2.8.1", "typing-extensions>=4.11; python_version < \"3.11\""]
```

Comment:
There are no changes required in the Pulumi Python SDK to support TypedDicts, the changes are purely in the Provider SDKs. The reason for this is that TypedDicts are purely a typechecking concept and don't exist at runtime in Python. Previously you could already pass in dictionaries instead of args classes, but these were always untyped. Now they are typed.

Previously inputs looked like his

```python
# pulumi.InputType = Union[T, Mapping[str, Any]]

Optional[pulumi.Input[pulumi.InputType['AnalyzerConfigurationArgs']]]
```
and with this change, we essentially inline the `InputType` Union with a stricter type:
```python
# class AnalyzerConfigurationArgsDict(TypedDict):
#    unused_access: NotRequired[pulumi.Input['AnalyzerConfigurationUnusedAccessArgsDict']]
# ...

Optional[pulumi.Input[Union['AnalyzerConfigurationArgs', 'AnalyzerConfigurationArgsDict']]]
```

At runtime, when you pass in a concrete instance of a dictionary for an input, it is just a standard `dict`.